src/main/java/fi/metatavu/metamind/bot/tokenizer/SimpleTokenizer.java

```diff
@@ -0,0 +1,132 @@
+package fi.metatavu.metamind.bot.tokenizer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.validator.routines.EmailValidator;
+
+import com.rabidgremlin.mutters.core.Tokenizer;
+
+/**
+ * THIS IS A COPY OF ORIGINAL CLASS. REMOVE WHEN MUTTERS SUPPORTS FINNISH CHARACTERS ON TOKENIZATION
+ */
+
+/**
+ * This class implements a 'simple' tokenizer that splits on whitespace and strips out punctuation.
+ * 
+ * It does attempt to preserve anything that looks like a decimal number, date, time, email address or a templated
+ * utterance slot tag.
+ * 
+ * It also supports lower casing of tokens which can help with intent matching.
+ * 
+ * @author rabidgremlin
+ *
+ */
+public class SimpleTokenizer
+    implements Tokenizer
+{
+  /** Are we forcing tokens to lower case ? */
+  private boolean forceLowerCase;
+
+  /**
+   * Constructor. Defaults to not lowercasing tokens.
+   */
+  public SimpleTokenizer()
+  {
+    this(false);
+  }
+
+  /**
+   * Constructor.
+   * 
+   * @param forceLowerCase Set to true if tokens should be lower cased.
+   */
+  public SimpleTokenizer(boolean forceLowerCase)
+  {
+    this.forceLowerCase = forceLowerCase;
+  }
+
+  @Override
+  public String[] tokenize(String text)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 40 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)